packages/lesswrong/components/posts/usePostReadProgress.ts

```diff
@@ -27,50 +28,106 @@ interface UsePostReadProgressProps {
   setScrollWindowHeight?: (element: HTMLDivElement, height: number) => void;
 
   /**
-   * Progress bars in the full-height ToC which display a sliding window need to include the height of the initial viewport when calculating the scroll percentage
+   * Whether to use the fixed ToC scroll calculation method.
+   * If true, uses the method that calculates based on viewport bottom relative to post top (viewport-centric).
+   * If false, uses the original method that calculates based on post bottom position (post-centric).
    */
-  useFirstViewportHeight?: boolean;
+  useFixedToCScrollCalculation?: boolean;
 }
 
-const clampPct = getClamper(0, 1);
+const clampPct = getClamper(0, 100);
 
-export const usePostReadProgress = ({ updateProgressBar, disabled = false, delayStartOffset = 0, setScrollWindowHeight, useFirstViewportHeight = false }: UsePostReadProgressProps) => {
+export const usePostReadProgress = ({ 
+  updateProgressBar, 
+  disabled = false, 
+  delayStartOffset = 0, 
+  setScrollWindowHeight, 
+  useFixedToCScrollCalculation = true 
+}: UsePostReadProgressProps) => {
   const readingProgressBarRef = useRef<HTMLDivElement|null>(null);
 
-  const getScrollPct = (postBodyElement: HTMLElement, delayStartOffset: number, clamp = true) => {
+  /**
+   * Original method for calculating scroll percentage
+   * Based on post body bottom position relative to viewport
+   */
+  const getOriginalScrollPct = (postBodyElement: HTMLElement, offset: number, clamp = true): number => {
     // position of post body bottom relative to the bottom of the viewport
     const postBodyBottomPos = postBodyElement.getBoundingClientRect().bottom - window.innerHeight;
     // total distance from top of page to post body bottom
-    const totalHeight = useFirstViewportHeight ? postBodyElement.getBoundingClientRect().height : window.scrollY + postBodyBottomPos;
+    const totalHeight = window.scrollY + postBodyBottomPos;
   
-    const scrollPercent = 1 - ((postBodyBottomPos + delayStartOffset) / totalHeight);
+    const scrollPercent = 1 - ((postBodyBottomPos + offset) / totalHeight);
     const adjustedScrollPercent = clamp ? clampPct(scrollPercent) : scrollPercent;
-    return adjustedScrollPercent * 100;
+    return adjustedScrollPercent;
+  }
+
+  /**
+   * New method for calculating scroll percentage
+   * Calculates what percentage of the post has been scrolled through
+   * Uses the bottom of the viewport as the reference point
+   * @returns A value between 0-100 representing percentage of post that's been scrolled, unless clamp=false
+   */
+  function getFixedToCScrollPct(
```

Comment:
for consistency's sake this should probably be a const - in practice I don't think it'll matter but I have ever run into execution order issues when mixing functions and const functions, since regular functions are hoisted: https://developer.mozilla.org/en-US/docs/Glossary/Hoisting